### PR TITLE
Horizon

### PR DESCRIPTION
**Pull request by Garyfallidis**
DIPY Horizon is a tool based on FURY for visualizing medical data.

Garyfallidis E., M-A. Côté, B.Q. Chandio, S. Fadnavis, J. Guaje, R. Aggarwal, E. St-Onge, K.S. Juneja, S. Koudoro, D. Reagan, DIPY Horizon: fast, modular, unified and adaptive visualization, Proceedings of: International Society of Magnetic Resonance in Medicine (ISMRM), Montreal, Canada, 2019.

### CODE REVIEW

File: dipy/viz/app.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+    length_gt : float
+    clusters_lt : int
+    clusters_gt : int
+    world_coords : bool
+    interactive : bool
+    out_png : string
+
+    References
+    ----------
+    .. [Horizon_ISMRM19] Garyfallidis E., M-A. Côté, B.Q. Chandio,
```
Comment (skoudoro) [inline dipy/viz/app.py]: Hi @Garyfallidis, you still have an ASCII issue here

File: dipy/viz/app.py [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+        self.world_coords = world_coords
+        self.interactive = interactive
+        self.prng = np.random.RandomState(27)
+        self.tractograms = tractograms
+        self.out_png = out_png
+
+        if images is None:
+            self.images = []
+        else:
+            self.images = images
```
Comment (skoudoro) [inline dipy/viz/app.py]: I think you can use the pythonic way `self.images = images or []` instead of `if/else` directive

File: dipy/viz/app.py [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+        scene = window.Renderer()
+        for (t, streamlines) in enumerate(self.tractograms):
+            if self.random_colors:
+                colors = self.prng.random_sample(3)
+            else:
+                colors = None
+
+            if self.cluster:
+
+                # _cluster(streamlines, cluster_thr, )
```
Comment (arokem) [inline dipy/viz/app.py]: Remove commented code.

File: dipy/workflows/viz.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                tractograms.append(streamlines)
+
+            if ends('.nii.gz') or ends('.nii'):
+
+                data, affine = load_nifti(f)
+                images.append((data, affine))
+                if verbose:
+                    print(affine)
+
+            """ TODO: add support for peaks
```
Comment (arokem) [inline dipy/workflows/viz.py]: Is this something you want to do in this PR? 

If not, I suggest removing this code and posting an issue to keep track of this.

File: dipy/workflows/viz.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        tractograms = []
+        images = []
+        interactive = not stealth
+        world_coords = not native_coords
+
+        io_it = self.get_io_iterator()
+
+        for input_output in io_it:
+
+            f = input_output[0]
```
Comment (arokem) [inline dipy/workflows/viz.py]: Best to avoid single letter variable names, because they are hard to search for.

**Comment by Garyfallidis** [comment]
Run this workflow from the command line in the following way
```
        >>> dipy_horizon streamlines.trk brain.nii.gz

        or

        >>> dipy_horizon streamlines.tck --cluster --cluster_thr 15

        or if you do not want visual output use

        >>> dipy_horizon fa.nii.gz --stealth
```

**Comment by arokem** [comment]
Thanks! I'll give it a whirl. 

Do you want to add a page to the documentation with these instructions?

**Comment by arokem** [comment]
What does it mean that it doesn't have visual output in `stealth` mode? That it just creates a figure? 

Also, I get an output the terminal, like this: 

```
(dipy) sess-01$dipy_horizon dwi_dti_fa.nii.gz
Loading file ...
dwi_dti_fa.nii.gz


[[-1.74415660e+00  1.30238131e-01  7.89432004e-02  1.07013458e+02]
 [ 1.37312442e-01  1.72685742e+00  3.33311886e-01 -1.19927002e+02]
 [ 3.95377614e-02 -2.51994908e-01  2.32490230e+00 -2.33210659e+01]
 [ 0.00000000e+00  0.00000000e+00  0.00000000e+00  1.00000000e+00]]
```

What does that matrix mean? Would be good to provide more information.

**Comment by Garyfallidis** [comment]
We can add docs later. For now I need this piece merged to support the workshop. After that will work on improving coverage, adding new awesome features and make beautiful docs online. There is plenty of work todo. But we have two important deadlines a) NIH report, b) awesome DIPY workshop! I suggest we just merge for now.

**Comment by arokem** [comment]
OK. Let's do that. It's looking really good, but there is also some functionality that doesn't work on my end. I'll post issues after merging. 

Given that, I think this is ready to go (all @skoudoro's comments are addressed, right?). If no one else shouts loudly, I will merge in a couple of days.

**Pull request merged**